torchvision/datapoints/_datapoint.py

```diff
@@ -25,9 +32,13 @@ def _to_tensor(
             requires_grad = data.requires_grad if isinstance(data, torch.Tensor) else False
         return torch.as_tensor(data, dtype=dtype, device=device).requires_grad_(requires_grad)
 
+    @classmethod
+    def _wrap(cls: Type[D], tensor: torch.Tensor) -> D:
+        return tensor.as_subclass(cls)
+
     @classmethod
     def wrap_like(cls: Type[D], other: D, tensor: torch.Tensor) -> D:
-        raise NotImplementedError
+        return cls._wrap(tensor)
```

Comment:
Wait, so now I'm a bit confused. 
We clearly don't use `other` so we should remove it from there.


BUT then:

-  don't we want to allow `wrap_like(other, t)` so that t gets the same device (and dtype??) as `other`??  we don't seem to do that
- do we eve need `wrap_like` for those datapoints that don't have meta-data? What's the point of doing `Image.wrap_like(other, t)` or even `Image.wrap_like(t)` (not supported) when one can just do `Image(t)`?

Anyway, I'd suggest to keep things as-is for now, and to address that stuff after this PR is merged.

Reply:
The reason we have `wrap_like` the way it is, is that we use it inside `__torch_function__`:

https://github.com/pytorch/vision/blob/9ebf10af02aeb882bd8d7782149e21b48528d562/torchvision/datapoints/_datapoint.py#L32-L39

Basically, we need a generic way for "Here is a datapoint input possibly with some metadata. Given a plain tensor, make it the same type as the datapoint and copy the metadata if available".

So the reason we have `other` in all of them is because if we need metadata and pass nothing explicitly, we need to take it from `other`. This also means that on datapoints that have no metadata, we never use `other`.

Reply:
OK. I understand this is useful for us internally but I'm still questioning whether we should expose that as public for datapoints that don't have meta-data. Let's just keep things as-is in this PR so we can merge it soon, and revisit just after.